### PR TITLE
nx2elf: init at unstable-2020-05-26

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
Due to security concerns, I've updated the (4 year old) local version of `lz4` with the one from nixpkgs, which incidentally fixes a warning.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/compression/nx2elf/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchFromGitHub {
+    owner = "shuffle2";
+    repo = "nx2elf";
+    rev = "7212e82a77b84fcc18ef2d050970350dbf63649b";
+    sha256 = "1j4k5s86c6ixa3wdqh4cfm31fxabwn6jcjc6pippx8mii98ac806";
+  };
+
+  buildInputs = [ lz4 ];
+
+  preBuild = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/nx2elf/default.nix:16]:
```suggestion
  postPatch = ''
```

Comment (IvarWithoutBones) [reply pkgs/tools/compression/nx2elf/default.nix]: Done :+1:

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119493` at b8f93bd9 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>nx2elf</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/compression/nx2elf/default.nix:25:3:
```
   |
25 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by IvarWithoutBones** [comment]
> Result of `nixpkgs-review pr 119493` at [b8f93bd](https://github.com/NixOS/nixpkgs/commit/b8f93bd9c808e6a7bd868494e0285234563cd8ee) run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package built successfully:
> 1 suggestion:
> Result of `nixpkgs-review pr 119493` at [b8f93bd](https://github.com/NixOS/nixpkgs/commit/b8f93bd9c808e6a7bd868494e0285234563cd8ee) run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package failed to build:
> * [nx2elf](https://gist.github.com/5bc1cb3b1ce038b068acf554e04b77bc)
> 
> Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
> Please exercise your independent judgement.

I've set the platform to `x86_64-linux` for now, as I do not have a Darwin machine to test this on.

File: pkgs/tools/compression/nx2elf/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  installPhase = ''
+    mkdir -p $out/bin
+    install -D nx2elf $out/bin/nx2elf
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/shuffle2/nx2elf";
+    description = "Convert Nintendo Switch executable files to ELFs";
+    license = licenses.unfree; # No license specified upstream
+    platforms = [ "x86_64-linux " ]; # Should work on Darwin as well, but this is untested. aarch64-linux fails.
```
Comment (fortuneteller2k) [inline pkgs/tools/compression/nx2elf/default.nix:34]:
```suggestion
    platforms = [ "x86_64-linux" ]; # Should work on Darwin as well, but this is untested. aarch64-linux fails.
    broken = stdenv.hostPlatform.isAarch64;
```

**Comment by SuperSandro2000** [comment]
@ofborg build nx2elf

File: pkgs/tools/compression/nx2elf/default.nix (hunk 2 above)
Comment (SuperSandro2000) [inline pkgs/tools/compression/nx2elf/default.nix:34]:
```suggestion
    platforms = [ "x86_64-linux" ]; # Should work on Darwin as well, but this is untested. aarch64-linux fails.
```

**Pull request merged**